From bdbf728999001e72f061aaf61097b7af3d01af14 Mon Sep 17 00:00:00 2001
From: mcz <emcze@ya.ru>
Date: Sun, 21 Oct 2018 19:18:01 +0200
Subject: [PATCH] bt: fix bluealsa pcm device can't found [1/2]

PD#SH-652

Problem:
With the release of alsa-lib v1.1.7, the location for add-on configs has
changed to /etc/alsa/conf.d/

Solution:
cherry-pick from upstream:dae069984bcae46004f8f3643fa0b14fb24f994f
Move ALSA add-on config file to /etc/alsa/conf.d
For alsa <1.1.7 backwards compatibility is preserved.

Verify:
ad401

Change-Id: If04839131f6a5e2ed1edf781324d99c5aea7ac32
Signed-off-by: xiang.lin <xiang.lin@amlogic.com>
---
 configure.ac           | 18 +++++++++++-------
 src/asound/Makefile.am |  4 ++--
 2 files changed, 13 insertions(+), 9 deletions(-)

diff --git a/configure.ac b/configure.ac
index 60d3807..fcddab4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -47,6 +47,7 @@ PKG_CHECK_MODULES([GIO2], [gio-unix-2.0])
 PKG_CHECK_MODULES([SBC], [sbc >= 1.2])
 
 AM_CONDITIONAL([ALSA_1_1_2], [$PKG_CONFIG --atleast-version=1.1.2 alsa])
+AM_CONDITIONAL([ALSA_1_1_7], [$PKG_CONFIG --atleast-version=1.1.7 alsa])
 
 AC_ARG_ENABLE([aac],
 	[AS_HELP_STRING([--enable-aac], [enable AAC support])])
@@ -92,25 +93,28 @@ AM_CONDITIONAL([ENABLE_PCM_TEST], [test "x$enable_pcm_test" = "xyes"])
 AC_ARG_WITH([alsaplugindir],
 	AS_HELP_STRING([--with-alsaplugindir=dir], [path where ALSA plugin files are stored]),
 	[alsaplugindir="$withval"], [alsaplugindir=$($PKG_CONFIG --variable=libdir alsa)/alsa-lib])
-AC_ARG_WITH([alsadatadir],
-	AS_HELP_STRING([--with-alsadatadir=dir], [directory containing ALSA data files]),
-	[alsadatadir="$withval"], [alsadatadir="$datadir/alsa"])
+AC_ARG_WITH([alsaconfdir],
+	AS_HELP_STRING([--with-alsaconfdir=dir], [directory containing ALSA add-on configuration files]),
+	[alsaconfdir="$withval"],
+	[AM_COND_IF([ALSA_1_1_7],
+		[alsaconfdir="$sysconfdir/alsa/conf.d"],
+		[alsaconfdir="$datadir/alsa/alsa.conf.d"])])
 
 test "x$prefix" = xNONE && prefix=$ac_default_prefix
 test "x$exec_prefix" = xNONE && exec_prefix=$prefix
 
 # TODO: Get rid of "ev(a|i)l" statements.
 # TIP: Wizard-level Autotconf coder is needed.
-eval alsadatadir="$alsadatadir"
-eval alsadatadir="$alsadatadir"
+eval alsaconfdir="$alsaconfdir"
+eval alsaconfdir="$alsaconfdir"
 eval alsaplugindir="$alsaplugindir"
 eval alsaplugindir="$alsaplugindir"
 
-AC_DEFINE_UNQUOTED([ALSA_DATA_DIR], "$alsadatadir", [Directory containing ALSA data files.])
+AC_DEFINE_UNQUOTED([ALSA_CONF_DIR], "$alsaconfdir", [Directory containing ALSA add-on configuration files.])
 AC_DEFINE_UNQUOTED([ALSA_PLUGIN_DIR], "$alsaplugindir", [Directory containing ALSA add-on modules.])
 AC_DEFINE_UNQUOTED([RUN_STATE_DIR], "$runstatedir", [Path where run statuses are stored.])
 
-AC_SUBST([ALSA_DATA_DIR], [$alsadatadir])
+AC_SUBST([ALSA_CONF_DIR], [$alsaconfdir])
 AC_SUBST([ALSA_PLUGIN_DIR], [$alsaplugindir])
 AC_SUBST([RUN_STATE_DIR], [$runstatedir])
 
diff --git a/src/asound/Makefile.am b/src/asound/Makefile.am
index 75a2dc3..923d884 100644
--- a/src/asound/Makefile.am
+++ b/src/asound/Makefile.am
@@ -5,7 +5,7 @@ EXTRA_DIST = 20-bluealsa.conf
 
 asound_module_ctl_LTLIBRARIES = libasound_module_ctl_bluealsa.la
 asound_module_pcm_LTLIBRARIES = libasound_module_pcm_bluealsa.la
-asound_module_data_DATA = 20-bluealsa.conf
+asound_module_conf_DATA = 20-bluealsa.conf
 
 libasound_module_ctl_bluealsa_la_SOURCES = \
 	../shared/ctl-client.c \
@@ -19,7 +19,7 @@ libasound_module_pcm_bluealsa_la_SOURCES = \
 
 asound_module_ctldir = @ALSA_PLUGIN_DIR@
 asound_module_pcmdir = @ALSA_PLUGIN_DIR@
-asound_module_datadir = @ALSA_DATA_DIR@/alsa.conf.d
+asound_module_confdir = @ALSA_CONF_DIR@
 
 AM_CFLAGS = \
 	-I$(top_srcdir)/src \
-- 
2.29.0

